app/javascript/packages/document-capture/components/acuant-capture.jsx

```diff
@@ -69,6 +46,28 @@ const DEFAULT_ACCEPTABLE_FILE_SIZE_BYTES =
     ? 250 * 1024
     : Number(process.env.ACUANT_MINIMUM_FILE_SIZE);
 
+/**
+ * Returns an instance of File representing the given data URL.
+ *
+ * @param {string} dataURL Data URL.
+ *
+ * @return {Blob} File representation.
+ */
+function toBlob(dataURL) {
+  const [header, data] = dataURL.split(',');
+  const isBase64 = /;base64$/.test(header);
+  const [type] = header.replace(/^data:/, '').split(';');
+  const decodedData = isBase64 ? window.atob(data) : decodeURIComponent(data);
+
+  const buffer = new ArrayBuffer(decodedData.length);
+  const view = new Uint8Array(buffer);
+  for (let i = 0; i < decodedData.length; i++) {
+    view[i] = decodedData.charCodeAt(i);
+  }
+
+  return new window.Blob([buffer], { type });
```

Comment:
Doing loops of manual data conversion always makes me pause (since usually library functions can do it faster), especially on large files.

I poked around at the docs.... looks like like TypedArray.from can take a conversion callback, so as far as I can tell, this works:


```suggestion
  const arr = Uint8Array.from(decodedData, c => c.charCodeAt(0));
  return new window.Blob([arr], { type });
```

If I'm reading the docs correctly, [`Blob`](https://developer.mozilla.org/en-US/docs/Web/API/Blob/Blob) can take an [`ArrayBufferView`](https://developer.mozilla.org/en-US/docs/Web/API/ArrayBufferView) which I *think* means we could pass in a Uint8Array directly?

Reply:
> Doing loops of manual data conversion always makes me pause (since usually library functions can do it faster), especially on large files.

Yep, you're right, `Uint8Array.from` is faster 👍  (almost 2x!)

https://jsbin.com/fohicec/2/edit?html,js,console,output

Reply:
A wild `TextEncoder` appears (10x faster than `for`, 6x faster than `Uint8Array.from`):

https://jsbin.com/fohicec/3/edit?html,js,console,output

Reply:
Wow winner!!

Reply:
Testing on a real device, `TextEncoder` didn't appear to work correctly. I'm guessing it might be related to the fact that it encodes as UTF-8, where the other two implementations would not (?). There _might_ be some way to get it to play nice. I tried a few ideas with no avail (e.g. setting `;charset=UTF-8` in `type` parameter). Could be worth revisiting later, but for the meantime, I'll stick with your original suggestion.